### ISSUE TITLE
[Bug]: Root error selectSelectedInternalAccount being displayed after deleting hardware account

### ISSUE THREAD

**Issue by vivek-consensys** [MetaMask/metamask-mobile#9786]
### Describe the bug

Error appears after deleting hardware wallet account, when clicking 'try again' button. there are no issues.

### Expected behavior

No errors after deleting hardware wallet account

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/106310394/8ca1585d-584a-4ab8-9570-3c84a1352ba8

### Steps to reproduce

1. install MetaMask 7.23.0 RC5
2. import from SRP
3. proceed to accounts
4. add hardware wallet
5. connect Ledger / QR based account
6. verify the account is added
7. delete hardware account from account list
8. error is displayed

### Error messages or log output

```shell
View: Root
Error: selectSelectedInternalAccount: Account with ID cc51cc3f-4506-4cfd-808b-388de93b1e70 not found.
The error may be correlated with this previous error:
construct@[native code]
r@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:71:62926
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:6065:512
E@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3335:564
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3334:1041
E@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3335:564
t@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1132:848
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:502
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:899
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:46:827
notify@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:861
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:427
C@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1115:2995
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:502
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:899
Pe@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:255:98935
Ce@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:255:14116
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:46:827
notify@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:861
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:427
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:1111:502
b@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3354:1611
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3339:10440
C@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3460:2392
v@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3803:1514
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3803:1630
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3804:2420
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3802:1651
h@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3905:8330
c@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3905:6511
v@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3803:1514
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3803:1630
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3804:2420
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3802:1651
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3945:20881
l@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:3945:999
u@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:94:158
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:94:870
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:100:1471
T@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:100:419
p@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:100:692
callReactNativeMicrotasks@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:100:2871
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:37:2875
@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:37:966
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:37:2511
value@/private/var/containers/Bundle/Application/F3D92727-C760-4084-8242-30F8744E3675/MetaMask.app/main.jsbundle:37:924
```

### Version

7.23.0 RC5

### Build type

Beta

### Device

iPhone 13

### Operating system

iOS

### Severity

sev-2

**Comment by dawnseeker8** [comment]
unable to replicate the issue in both `main` branch and `7.24.0` branch, and i have tested both `android 23` and `ios17`. issue can't be replicated.

maybe it has been fixed after 7.23 RC5 release.

**Comment by vivek-consensys** [comment]
Unable to reproduce using 7.24.0 RC2 (1335), closing issue.

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNGhYQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--31d77a4736c70942ae59b87e619f4d634f6003dc/RPReplay_Final1718179478.MP4)
